### PR TITLE
Python: Shared dataflow, parameter routing tests

### PR DESCRIPTION
**Pull request by yoff**
This PR adds tests of parameter routing. The actual query should be rewritten at some point. In its current form, it is unlikely to find calls (and thus routes) not looking syntactically like function calls or method calls.

It should be decided if this complication of the calls tests is fine or if they should be made in a copy of classes.py.

### CODE REVIEW

File: python/ql/test/experimental/dataflow/coverage/argumentRouting1.ql [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+ * A configuration to check routing of arguments through magic methods.
+ */
+class ArgumentRoutingConfig extends DataFlow::Configuration {
+  ArgumentRoutingConfig() { this = "ArgumentRoutingConfig" }
+
+  override predicate isSource(DataFlow::Node node) {
+    exists(AssignmentDefinition def |
+      def.getVariable() = node.(DataFlow::EssaNode).getVar() and
+      def.getValue().(DataFlow::DataFlowCall).getCallable().getName().matches("With\\_%")
+    ) and
```
[the remark this answers is not in the document]
Comment (yoff) [reply python/ql/test/experimental/dataflow/coverage/argumentRouting1.ql:13]: We are looking for the construction `with_add = With_add()`. If we see that, we want `with_add` to be a source. Without the conjunct, we would get all occurrences of `with_add` as sources.
We could take a different stance and say that the call to `With_add` is the source, as that is where the object with the special method is created..

[the remark this answers is not in the document]
Comment (tausbn) [reply python/ql/test/experimental/dataflow/coverage/argumentRouting1.ql:13]: Ah, I was thinking we could just look for `with_add = ...`, and not care about what we assign to the variable (i.e. simply make it a source if it has an assignment with a name with `with_` as a prefix). 

(In fact, aren't you getting every instance of `with_add` anyway? It looks to me as if the above conjunct indeed looks for an assignment of `with_add` of the right shape, but it doesn't constrain `node` to be _that_ occurrence of `with_add`, but perhaps I'm misreading things.)

[the remark this answers is not in the document]
Comment (yoff) [reply python/ql/test/experimental/dataflow/coverage/argumentRouting1.ql:13]: I think line 11 makes that restriction?
(Also, as far as I recall, quick-eval confirmed this.)

[the remark this answers is not in the document]
Comment (yoff) [reply python/ql/test/experimental/dataflow/coverage/argumentRouting1.ql:13]: Indeed quick-eval does confirm this, e.g. there are only two sources for `with_bool` (one for each function).

**Review by tausbn — APPROVED** [review]

**Pull request merged**